File: cpp/Map/MapScriptBaseMaterial.c
// Type: Map.MapScriptBaseMaterial
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Map/MapScriptBaseMaterial.cs
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/Materials/MapScriptBaseMaterial.cs
// --------------------------------

// Map.MapScriptBaseMaterial$$.ctor
// il2cpp: void Map_MapScriptBaseMaterial___ctor (Map_MapScriptBaseMaterial_o* __this, const MethodInfo* method);
// 0x40b5f30

void Map_MapScriptBaseMaterial___ctor(Map_MapScriptBaseMaterial_o *__this,MethodInfo *method)

{
  Utility_Color255_o *__this_00;
  
  if (g_data_057ac4ed == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_Color255);
    il2cpp_runtime_helper_023445d0(&"Default");
    g_data_057ac4ed = '\x01';
  }
  (__this->fields).Shader = "Default";
  il2cpp_runtime_helper_022b4080(&__this->fields);
  __this_00 = (Utility_Color255_o *)il2cpp_runtime_helper_023052d0(TypeInfo_Color255);
  Utility_Color255___ctor_4388b90(__this_00,(MethodInfo *)0x0);
  (__this->fields).Color = __this_00;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Color);
  Utility_BaseCSVRowItem___ctor((Utility_BaseCSVRowItem_o *)__this,(MethodInfo *)0x0);
  return;
}

File: csharp_body_deref/Scripts/Map/MapScriptBaseMaterial.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Map.MapScriptBaseMaterial
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Map/MapScriptBaseMaterial.c
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/Materials/MapScriptBaseMaterial.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;
using Utility;

namespace Map;

[Token(Token = "0x200027D")]
public class MapScriptBaseMaterial : BaseCSVRowItem
{
	[Token(Token = "0x4000D8B")]
	[FieldOffset(Offset = "0x10")]
	[Order(1)]
	public string Shader;

	[Token(Token = "0x4000D8C")]
	[FieldOffset(Offset = "0x18")]
	[Order(2)]
	public Color255 Color;

	[Token(Token = "0x6000F4F")]
	[Address(RVA = "0x3FB5F30", Offset = "0x3FB5F30", VA = "0x3FB5F30")]
	public MapScriptBaseMaterial()
	{
	}
}
